### PR TITLE
Add a function to get the maximum width type for a given numeric type

### PR DESCRIPTION
**Pull request by cpcloud**

### CODE REVIEW

**Comment by cpcloud** [comment]
@mrocklin any comments here? this is needed for ContinuumIO/blaze#925

File: datashape/typesets.py [hunk 1]
```diff
@@ -115,3 +120,70 @@ def __getitem__(self, key):
...
+    >>> supertype(float32)
+    {floating}
+
+    >>> supertype(complex128)
+    {complexes}
+
+    >>> supertype(bool_)
+    {boolean}
+    """
+    measure = getattr(measure, 'ty', measure)
```
Comment (mrocklin) [inline datashape/typesets.py]: This seems easier to understand.  Actually, I'm not even sure why `ty` is the field name for this.  What does that stand for?
```
if isinstance(measure, Option):
    measure = measure.ty
```

File: datashape/typesets.py [hunk 2]
```diff
@@ -115,3 +120,70 @@ def __getitem__(self, key):
...
+    uint16: unsigned,
+    uint32: unsigned,
+    uint64: unsigned,
+    float16: floating,
+    float32: floating,
+    float64: floating,
+    complex64: complexes,
+    complex128: complexes,
+    bool_: boolean
+}
```
Comment (mrocklin) [inline datashape/typesets.py]: We might want to store these on the types themselves.

**Comment by mrocklin** [comment]
I made a couple of nitpicky comments but they could probably be ignored.  +1

File: datashape/typesets.py (hunk 1 above)
Comment (cpcloud) [inline datashape/typesets.py]: Not sure where this nomenclature comes from. I think it means `ty`pe

Comment (cpcloud) [inline datashape/typesets.py]: maybe @mwiebe knows?

Comment (mwiebe) [inline datashape/typesets.py]: It means dtype, basically. I think we discussed this a few times, and decided to get rid of the "measure" terminology, but nobody has acted on that decision within the datashape codebase.

File: datashape/typesets.py (hunk 2 above)
Comment (cpcloud) [inline datashape/typesets.py]: i'm going to leave these as is for now

**Pull request merged**